googleapiclient/discovery.py

```diff
@@ -945,24 +954,30 @@ def methodNext(self, previous_request, previous_response):
     # Retrieve nextPageToken from previous_response
     # Use as pageToken in previous_request to create new request.
 
-    if 'nextPageToken' not in previous_response or not previous_response['nextPageToken']:
+    if nextPageToken not in previous_response or not previous_response[nextPageToken]:
       return None
 
     request = copy.copy(previous_request)
 
-    pageToken = previous_response['nextPageToken']
-    parsed = list(urlparse(request.uri))
-    q = parse_qsl(parsed[4])
-
-    # Find and remove old 'pageToken' value from URI
-    newq = [(key, value) for (key, value) in q if key != 'pageToken']
-    newq.append(('pageToken', pageToken))
-    parsed[4] = urlencode(newq)
-    uri = urlunparse(parsed)
-
-    request.uri = uri
-
-    logger.info('URL being requested: %s %s' % (methodName,uri))
+    token = previous_response[nextPageToken]
+
+    if isPageTokenParameter:
+        # Replace pageToken value in URI
+        parsed = list(urlparse(request.uri))
```

Comment:
Yep --- this was old code; seemed oddly low-level here, but I didn't search the oauth2 library.